pkg/controller/operconfig/ingressconfig_controller.go

```diff
@@ -0,0 +1,96 @@
+package operconfig
+
+import (
+	"context"
+	"log"
+
+	operv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-network-operator/pkg/names"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// newIngressConfigReconciler returns a new reconcile.Reconciler
+func newIngressConfigReconciler(client client.Client) *ReconcileIngressConfigs {
+	return &ReconcileIngressConfigs{client: client}
+}
+
+var _ reconcile.Reconciler = &ReconcileIngressConfigs{}
+
+// ReconcileIngressConfigs watches for updates to ingress controller configuration
+// and sets the network policy related labels on the openshift-host-network namespace
+type ReconcileIngressConfigs struct {
+	client client.Client
+}
+
+// Reconcile sets the openshift-host-network namespaces' labels as per the
+// endpointPublishingStrategy of the `default` ingress controller object.
+func (r *ReconcileIngressConfigs) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	if request.Namespace == names.IngressControllerNamespace && request.Name == names.DefaultIngressControllerName {
+		log.Printf("Reconciling update to %s/%s\n", request.Namespace, request.Name)
+		ingressControllerConfig := &operv1.IngressController{TypeMeta: metav1.TypeMeta{APIVersion: operv1.GroupVersion.String(), Kind: "IngressController"}}
+		err := r.client.Get(context.TODO(), request.NamespacedName, ingressControllerConfig)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Printf("Ingress Controller configuration %s was deleted", request.NamespacedName.String())
+				// Request object not found, could have been deleted after reconcile request.
+				// Owned objects are automatically garbage collected, since we set
+				// the ownerReference (see https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/).
+				// Return and don't requeue
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - requeue the request.
+			log.Printf("Unable to retrieve IngressController.operator.openshift.io object: %v", err)
+			return reconcile.Result{}, err
+		}
+		labels := map[string]*string{names.PolicyGroupLabel: nil}
+		log.Printf("Ingress Controller Config: %+v", ingressControllerConfig)
```

Comment:
This log is probably unnecessary.

Reply:
Yeah I had added it for debugging, but it also helps keep it since we don't expect too many objects to be reconciled (esp. if something were to go cuckoo here in prod environments).